moneo.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import argparse
+import os
+
+def deploy(args):
+    dep_cmd='ansible-playbook -i ' + args.host_ini + ' src/ansible/deploy.yaml'
+    
+    if args.deploy == 'workers':
+        dep_cmd= dep_cmd + ' -e "skip_master=true"'
+    if args.deploy == 'master' :
+        dep_cmd= dep_cmd + ' -e "skip_worker=true"'
+    
+    print('Deployment type: ' + args.deploy)
+    os.system(dep_cmd)
+    
+def stop(args):
+    while(True) :
+        confirm = input("Are you sure you would like to stop Moneo? (Y/n)\n")
+        
+        if confirm.upper() == 'Y' :    
+            dep_cmd='ansible-playbook -i ' + args.host_ini + ' src/ansible/shutdown.yaml'
+            if args.shutdown=='workers':
+                dep_cmd= dep_cmd + ' -e "skip_master=true"'
+            if args.shutdown == 'manager':
+                dep_cmd= dep_cmd + ' -e "skip_worker=true"'
+            os.system(dep_cmd)
+            print("Moneo is Shutting down \n")
+            return 0
+        
+        elif confirm.upper() == 'N':
+            print("Canceling request to shutdown Moneo \n")
+            return 0
+        else:
+            print("Input not recognized\n")
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description='Moneo CLI',prog='moneo.py',usage='%(prog)s [-d {manager,workers,full}] [-c HOST_INI] \nusage: %(prog)s [-s {manager,workers,full}] [-c HOST_INI]')  
+    parser.add_argument('-d','--deploy' ,choices=['manager', 'workers', 'full'],help=' Deployment choices: {manager,workers,full}. Requires config file to be specified (i.e. -c host.ini) or file to be in Moneo directory.')
+    parser.add_argument('-s','--shutdown',choices=['manager', 'workers', 'full'], help='Shutdown choices: {manager,workers,full}.  Requires config file to be specified (i.e. -c host.ini) or file to be in Moneo directory.')
```

Comment:
It is good to add the flexibility that we shutdown moneo on specific hosts. But, it will be good to add a default option for shutting down on all hosts.

Also, how do we handle the case that user misses shutdown on specific hosts? Do we capture in subsequent moneo start operations?

Reply:
Right now the CLI will require an argument for shutdown/start up, so there are no default options.

In the event of subsequent start ups on the same host it will kill the current running exporters and start a new one. 

The way a user would miss a host is if they remove the host from the config file. In that event they would have to place the host back in the file to continue with any given start/shutdown operation. 

All of this is kind of how Moneo was initially designed. We may have to rework how we handle these cases in the future to make it more robust.

Reply:
This has been addressed. The CLI now has default option for complete deployment/shutdown. User can modify option if necessary.